classes/class-briqpay-confirmation.php

```diff
@@ -79,34 +79,34 @@ public function confirm_briqpay_order( $order_id ) {
 			return;
 		}
 
-		$session_id    = get_post_meta( $order_id, '_briqpay_session_id', true );
+		$session_id    = $order->get_meta( '_briqpay_session_id' );
 		$briqpay_order = BRIQPAY()->api->get_briqpay_order( array( 'session_id' => $session_id ) );
 
 		// Set post meta and complete order.
-		update_post_meta( $order_id, '_shipping_phone', $briqpay_order['shippingaddress']['cellno'] );
-		update_post_meta( $order_id, '_shipping_email', $briqpay_order['shippingaddress']['email'] );
-		update_post_meta( $order_id, '_briqpay_payment_method', $briqpay_order['purchasepaymentmethod']['name'] );
-		update_post_meta( $order_id, '_briqpay_psp_name', $briqpay_order['purchasepaymentmethod']['pspname'] );
-		update_post_meta( $order_id, '_briqpay_autocapture', $briqpay_order['purchasepaymentmethod']['autocapture'] );
-		update_post_meta( $order_id, '_briqpay_rules_result', wp_json_encode( $briqpay_order['rulesresult'] ?? array() ) );
-		update_post_meta( $order_id, '_billing_org_nr', $briqpay_order['orgnr'] );
+		$order->update_meta_data( '_shipping_phone', $briqpay_order['shippingaddress']['cellno'] );
+		$order->update_meta_data( '_shipping_email', $briqpay_order['shippingaddress']['email'] );
+		$order->update_meta_data( '_briqpay_payment_method', $briqpay_order['purchasepaymentmethod']['name'] );
+		$order->update_meta_data( '_briqpay_psp_name', $briqpay_order['purchasepaymentmethod']['pspname'] );
+		$order->update_meta_data( '_briqpay_autocapture', $briqpay_order['purchasepaymentmethod']['autocapture'] );
+		$order->update_meta_data( '_briqpay_rules_result', wp_json_encode( $briqpay_order['rulesresult'] ?? array() ) );
+		$order->update_meta_data( '_billing_org_nr', $briqpay_order['orgnr'] );
 
 		$purchase_payment_method = $briqpay_order['purchasepaymentmethod'];
 		if ( isset( $purchase_payment_method['pspSupportedOrderOperations'] ) ) {
 			if ( true === $purchase_payment_method['pspSupportedOrderOperations']['updateOrderSupported'] ) {
-				update_post_meta( $order_id, '_briqpay_psp_update_order_supported', $purchase_payment_method['pspSupportedOrderOperations']['updateOrderSupported'] );
+				$order->update_meta_data( '_briqpay_psp_update_order_supported', $purchase_payment_method['pspSupportedOrderOperations']['updateOrderSupported'] );
 			}
 		}
 
 		$order->set_payment_method_title( $briqpay_order['purchasepaymentmethod']['name'] );
 		$order->add_order_note( __( 'Payment via Briqpay, session ID: ', 'briqpay-for-woocommerce' ) . $session_id );
+		$order->save();
```

Comment:
Not sure, but should this be after the $order->payment_complete maybe? And if we do make a payment_complete, do we need the save call as well?

Reply:
`payment_complete` will also run `save` if it is successful, but only if it is successful. Therefore, we'll always need a `save`, and whether it is before or after does not really matter in this case.